students/ArchWu/Session2/series.py

```diff
@@ -0,0 +1,84 @@
+def fib(n):
+    """Return nth fibonacci number with recursion."""
+    if n == 0:
+        return 0
+    if n == 1:
+        return 1
+    return fib(n - 1) + fib (n - 2)
+
+def fib2(n):
+    """Another function that returns nth fibonacci number with lower time and space used."""
+    a, b = 0, 1
+    if n == 0:
+        return 0
+    for _ in range(n - 1):
+        a, b = b, a + b
+    return b
+
+def lucas(n):
+    """Return nth lucas number with low time and space complexity: O(n) time complexity and O(1) space complexity."""
+    a, b = 2, 1
+    if n == 0:
+        return 2
+    for _ in range(n - 1):
+        a, b = b, a + b
+    return b
+
+
+def sum_series(n, n0 = 0, n1 = 1):
+    """
+    compute the nth value of a summation series.
+
+    :param n0=0: value of zeroth element in the series
+    :param n1=1: value of first element in the series
+
+    This function should generalize the fibonacci() and the lucas(),
+    so that this function works for any first two numbers for a sum series.
+    Once generalized that way, sum_series(n, 0, 1) should be equivalent to fibonacci(n).
+    And sum_series(n, 2, 1) should be equivalent to lucas(n).
+    """
+    if n0 == 0 and n1 == 1:
+        return fib2(n)
+    elif n0 == 2 and n1 == 1:
+        return lucas(n)
+    else:
```

Comment:
Nice job! This else generalized logic should also be able to handle n0,n1 = 0,1 and n0,n1 = 2,1